### PR TITLE
Added ROATP Ver 2 tables

### PR DESCRIPTION
**Pull request by sarmaevanidfe**

### CODE REVIEW

File: src/SFA.DAS.DATA.MANAGEMENT/StoredProcedure/PopulateMetadataRolesAndPermissions.sql [hunk 1]
```diff
@@ -112,6 +112,17 @@ VALUES
 ('BetaUser','ASData_PL','MarketoSmartCampaigns','TABLE','SELECT',0,1),
 ('BetaUser','ASData_PL','PFBE_EmployerFeedback','TABLE','SELECT',0,1),
 ('BetaUser','ASData_PL','Resv_Reservation','TABLE','SELECT',0,1),
+('BetaUser','ASData_PL','RP_Apply','TABLE','SELECT',0,1),
+('BetaUser','ASData_PL','RP_Organisations','TABLE','SELECT',0,1),
+('BetaUser','ASData_PL','RP_Appeal','TABLE','SELECT',0,1),
+('BetaUser','ASData_PL','RP_AppealUpload','TABLE','SELECT',0,1),
+('BetaUser','ASData_PL','RP_AssessorPageReviewOutcome','TABLE','SELECT',0,1),
+('BetaUser','ASData_PL','RP_Contacts','TABLE','SELECT',0,1),
+('BetaUser','ASData_PL','RP_FinancialData','TABLE','SELECT',0,1),
```
[the remark this answers is not in the document]
Comment (sarmaevanidfe) [reply src/SFA.DAS.DATA.MANAGEMENT/StoredProcedure/PopulateMetadataRolesAndPermissions.sql]: Updated the scripts as per comments above

[the remark this answers is not in the document]
Comment (sarmaevanidfe) [reply src/SFA.DAS.DATA.MANAGEMENT/StoredProcedure/PopulateMetadataRolesAndPermissions.sql]: Logic amended for CreatedBy columns

File: src/SFA.DAS.DATA.MANAGEMENT/StoredProcedure/PopulateSourceDbMetadataForImport.sql [hunk 2]
```diff
@@ -179,13 +179,40 @@ VALUES
...
+		,'RP_Organisations')
+,('Apply','Appeal','dbo','[Id],[OversightReviewId],[CreatedOn]',''
+		,'[Message],[UserId],[UserName]',0,1
+		,'select [Id],[OversightReviewId],[Message],STUFF([UserId],2,len([UserId])-2,REPLICATE(''''*'''',len([UserId])-2)) As [UserId],STUFF([UserName],2,len([UserName])-2,REPLICATE(''''*'''',len([UserName])-2)) As [UserName],[CreatedOn] from [dbo].[Appeal]','RP_Appeal')
+,('Apply','AppealUpload','dbo','[Id],[ApplicationId],[AppealId],[FileStorageReference],[CreatedOn]',''
+		,'[Filename],[ContentType],[Size],[UserId],[UserName]',0,1
+		,'select [Id],[ApplicationId],[AppealId],[FileStorageReference],[Filename],[ContentType],[Size],STUFF([UserId],2,len([UserId])-2,REPLICATE(''''*'''',len([UserId])-2)) As [UserId],STUFF([UserName],2,len([UserName])-2,REPLICATE(''''*'''',len([UserName])-2)) As [UserName],[CreatedOn]  from [dbo].[AppealUpload]','RP_AppealUpload')
```
Comment (uhimabindu) [inline src/SFA.DAS.DATA.MANAGEMENT/StoredProcedure/PopulateSourceDbMetadataForImport.sql:194]: Might be good to add the same Checks to see if it is guid for all User Id Columns. In this case UserId.

File: src/SFA.DAS.DATA.MANAGEMENT/StoredProcedure/PopulateSourceDbMetadataForImport.sql [hunk 3]
```diff
@@ -179,13 +179,40 @@ VALUES
...
+,('Apply','AssessorPageReviewOutcome','dbo','[Id],[ApplicationId],[SequenceNumber],[SectionNumber],[PageId],[Assessor1ReviewStatus],[Assessor1ReviewComment],[Assessor2ReviewStatus],[Assessor2ReviewComment],[CreatedAt],[UpdatedAt]',''
+		,'[Assessor1UserId],[Assessor2UserId],[CreatedBy]',0,1
+		,'select [Id],[ApplicationId],[SequenceNumber],[SectionNumber],[PageId],STUFF([Assessor1UserId],2,len([Assessor1UserId])-2,REPLICATE(''''*'''',len([Assessor1UserId])-2)) As [Assessor1UserId],[Assessor1ReviewStatus],[Assessor1ReviewComment],STUFF([Assessor2UserId],2,len([Assessor2UserId])-2,REPLICATE(''''*'''',len([Assessor2UserId])-2)) As [Assessor2UserId],[Assessor2ReviewStatus],[Assessor2ReviewComment],[CreatedAt],case when TRY_CONVERT(UNIQUEIDENTIFIER,[CreatedBy]) IS NOT NULL Then [CreatedBy] Else  STUFF([CreatedBy],2,len([CreatedBy])-2,REPLICATE(''''*'''',len([CreatedBy])-2)) End As [CreatedBy],[UpdatedAt]  from [dbo].[AssessorPageReviewOutcome]','RP_AssessorPageReviewOutcome')
```
Comment (uhimabindu) [inline src/SFA.DAS.DATA.MANAGEMENT/StoredProcedure/PopulateSourceDbMetadataForImport.sql:197]: Might be good to add the same Checks to see if it is guid for all User Id Columns. Assessor1UserId and Assessor2UserId

File: src/SFA.DAS.DATA.MANAGEMENT/StoredProcedure/PopulateSourceDbMetadataForImport.sql [hunk 4]
```diff
@@ -179,13 +179,40 @@ VALUES
...
+		,'Select [Id],[ApplicationId],[SequenceNumber],[SectionNumber],[PageId],STUFF([ModeratorUserId],2,len([ModeratorUserId])-2,REPLICATE(''''*'''',len([ModeratorUserId])-2)) As [ModeratorUserId],STUFF([ModeratorUserName],2,len([ModeratorUserName])-2,REPLICATE(''''*'''',len([ModeratorUserName])-2)) As [ModeratorUserName],[ModeratorReviewStatus],[ModeratorReviewComment],STUFF([ClarificationUserId],2,len([ClarificationUserId])-2,REPLICATE(''''*'''',len([ClarificationUserId])-2)) As [ClarificationUserId],STUFF([ClarificationUserName],2,len([ClarificationUserName])-2,REPLICATE(''''*'''',len([ClarificationUserName])-2)) As [ClarificationUserName],[ClarificationStatus],[ClarificationComment],[ClarificationResponse],[ClarificationFile],[ClarificationUpdatedAt],[CreatedAt],case when TRY_CONVERT(UNIQUEIDENTIFIER,[CreatedBy]) IS NOT NULL Then [CreatedBy] Else  STUFF([CreatedBy],2,len([CreatedBy])-2,REPLICATE(''''*'''',len([CreatedBy])-2)) End As [CreatedBy],[UpdatedAt] from [dbo].[ModeratorPageReviewOutcome]','RP_ModeratorPageReviewOutcome')
```
Comment (uhimabindu) [inline src/SFA.DAS.DATA.MANAGEMENT/StoredProcedure/PopulateSourceDbMetadataForImport.sql:209]: Might be good to add the same Checks to see if it is guid for all User Id Columns. In this case ModeratorUserId

File: src/SFA.DAS.DATA.MANAGEMENT/StoredProcedure/PopulateSourceDbMetadataForImport.sql [hunk 5]
```diff
@@ -179,13 +179,40 @@ VALUES
...
+,('Apply','OversightReview','dbo','[Id],[ApplicationId],[GatewayApproved],[ModerationApproved],[Status],[ApplicationDeterminedDate],[InternalComments],[ExternalComments],[InProgressDate],[InProgressInternalComments],[InProgressExternalComments],[CreatedOn],[UpdatedOn]',''
+		,'[UserId],[InProgressUserId],[UserName],[InProgressUserName]',0,1
+		,'select [Id],[ApplicationId],[GatewayApproved],[ModerationApproved],[Status],[ApplicationDeterminedDate],[InternalComments],[ExternalComments],STUFF([UserId],2,len([UserId])-2,REPLICATE(''''*'''',len([UserId])-2)) As [UserId],STUFF([UserName],2,len([UserName])-2,REPLICATE(''''*'''',len([UserName])-2)) As [UserName],[InProgressDate],STUFF([InProgressUserId],2,len([InProgressUserId])-2,REPLICATE(''''*'''',len([InProgressUserId])-2)) As [InProgressUserId],STUFF([InProgressUserName],2,len([InProgressUserName])-2,REPLICATE(''''*'''',len([InProgressUserName])-2)) As [InProgressUserName],[InProgressInternalComments],[InProgressExternalComments],[CreatedOn],[UpdatedOn] from [dbo].[OversightReview]','RP_OversightReview')
```
Comment (uhimabindu) [inline src/SFA.DAS.DATA.MANAGEMENT/StoredProcedure/PopulateSourceDbMetadataForImport.sql:212]: Might be good to add the same Checks to see if it is guid for all User Id Columns. In this case UserId.

**Review by uhimabindu — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.DATA.MANAGEMENT/StoredProcedure/PopulateSourceDbMetadataForImport.sql (hunk 5 above)
Comment (uhimabindu) [reply src/SFA.DAS.DATA.MANAGEMENT/StoredProcedure/PopulateSourceDbMetadataForImport.sql:212]: Waiting for Prod Data to arrive to see how the User Id looks like with Live Data. Until then it's agreed to keep it masked to be safe. Will need revisiting once we have Live Data.

File: src/SFA.DAS.DATA.MANAGEMENT/StoredProcedure/PopulateSourceDbMetadataForImport.sql (hunk 4 above)
Comment (uhimabindu) [reply src/SFA.DAS.DATA.MANAGEMENT/StoredProcedure/PopulateSourceDbMetadataForImport.sql:209]: Waiting for Prod Data to arrive to see how the User Id looks like with Live Data. Until then it's agreed to keep it masked to be safe. Will need revisiting once we have Live Data.

File: src/SFA.DAS.DATA.MANAGEMENT/StoredProcedure/PopulateSourceDbMetadataForImport.sql (hunk 3 above)
Comment (uhimabindu) [reply src/SFA.DAS.DATA.MANAGEMENT/StoredProcedure/PopulateSourceDbMetadataForImport.sql:197]: Waiting for Prod Data to arrive to see how the User Id looks like with Live Data. Until then it's agreed to keep it masked to be safe. Will need revisiting once we have Live Data.

File: src/SFA.DAS.DATA.MANAGEMENT/StoredProcedure/PopulateSourceDbMetadataForImport.sql (hunk 2 above)
Comment (uhimabindu) [reply src/SFA.DAS.DATA.MANAGEMENT/StoredProcedure/PopulateSourceDbMetadataForImport.sql:194]: Waiting for Prod Data to arrive to see how the User Id looks like with Live Data. Until then it's agreed to keep it masked to be safe. Will need revisiting once we have Live Data.

**Review by uhimabindu — APPROVED** [review]
Approved

**Pull request merged**